src/screens/smart-screen/smart-screen.tsx

```diff
@@ -0,0 +1,200 @@
+import { smartConnect } from '../../core/utils/smart-connect';
+import { connect } from 'react-redux';
+import { Widgets } from '../../components/widgets/widgets';
+import { fetchScreenData } from '../../redux/ui/screens/data/actions';
+import { IScreenContext } from '../../components/widgets/types';
+import { IReduxState } from '../../redux/state';
+import { IScreenData, IScreenDatas } from '../../redux/ui/screens/data/state';
+import { withdraw, claimRewardNoInput } from '../../redux/wallets/actions/pos-actions';
+import { getScreenDataKey } from '../../redux/ui/screens/data/reducer';
+import { getSelectedAccount, getSelectedWallet } from '../../redux/wallets/selectors';
+import { getChainId } from '../../redux/preferences/selectors';
+import { IAccountState } from '../../redux/wallets/state';
+import { getAccountStats } from '../../redux/ui/stats/selectors';
+import { AccountStats } from '../../core/blockchain/types/stats';
+import { SkeletonPlaceholder } from '../../components/skeleton-placeholder/skeleton-placeholder';
+import stylesProvider from './styles';
+import { IThemeProps, withTheme } from '../../core/theme/with-theme';
+
+interface IExternalProps {
+    context: IScreenContext;
+}
+
+const mapStateToProps = (state: IReduxState) => {
+    const account = getSelectedAccount(state);
+    const chainId = account && getChainId(state, account.blockchain);
+
+    return {
+        dashboard: state.ui.screens.data.dashboard,
+        token: state.ui.screens.data.token,
+        accountStats:
+            account && getAccountStats(state, account.blockchain, chainId, account.address),
+        walletPublicKey: getSelectedWallet(state)?.walletPublicKey,
+        account,
+        chainId
+    };
+};
+
+interface IReduxProps {
+    dashboard: IScreenDatas;
+    token: IScreenDatas;
+
+    walletPublicKey: string;
+    account: IAccountState;
+    accountStats: AccountStats;
+    chainId: string;
+
+    fetchScreenData: typeof fetchScreenData;
+
+    claimRewardNoInput: typeof claimRewardNoInput;
+    withdraw: typeof withdraw;
+}
+
+const mapDispatchToProps = {
+    fetchScreenData,
+
+    claimRewardNoInput,
+    withdraw
+};
+
+interface IState {
+    screenData: IScreenData;
+    isLoading: boolean;
+    error: any;
+}
+
+export class SmartScreenComponent extends React.Component<
+    IReduxProps & IExternalProps & IThemeProps<ReturnType<typeof stylesProvider>>,
+    IState
+> {
+    constructor(
+        props: IReduxProps & IExternalProps & IThemeProps<ReturnType<typeof stylesProvider>>
+    ) {
+        super(props);
+        this.state = {
+            screenData: undefined,
+            isLoading: true,
+            error: undefined
+        };
+    }
+
+    public componentDidMount() {
+        this.props.fetchScreenData(this.props.context);
+    }
+
+    public componentDidUpdate(prevProps: IReduxProps & IExternalProps) {
+        if (
+            (this.props.account && this.props.account !== prevProps.account) ||
```

Comment:
make a hash here and compare it. its easier. check `ZilliqaTransactionUpdateComponent`  did update function